srcpkgs/systemd-boot/files/kernel.d/systemd-boot.post-install

```diff
@@ -0,0 +1,64 @@
+#!/bin/sh
+#
+# Kernel hook for systemd-boot.
+#
+# Arguments passed to this script: $1 pkgname, $2 version.
+#
+PKGNAME="$1"
+VERSION="$2"
+
+. "$ROOTDIR/etc/default/systemd-boot"
+
+if [ "$SYSTEMD_BOOT_DISABLE" ]; then
+	exit 0
+fi
+
+is_installed="$(bootctl is-installed)"
+
+if [ "$is_installed" != "yes" ]; then
+	echo "systemd-boot is not installed to the ESP"
+	echo "run 'bootctl install' to install it"
+	exit 0
+fi
+
+BOOT="${BOOT:-/boot}"
+ESP="${ESP:-"$BOOT"}"
+
+BOOT="$ROOTDIR/$BOOT"
+ESP="$ROOTDIR/$ESP"
+
+[ -d "$ESP" ] || exit 0
+
+ARCH="$(xbps-uhelper arch)"
+
+# EFI_ARCH is the identifier from the UEFI specification
+case "$ARCH" in
+	x86_64*) EFI_ARCH="x64";;
+	i686*) EFI_ARCH="IA32";;
+	aarch64*) EFI_ARCH="AA64";;
+esac
+
+entries="$ESP/loader/entries"
+name="void-$ARCH-$VERSION"
+entry="$entries/$name.conf"
+
+# check if user provided options in /etc/default/systemd-boot
+if [ -z "$CMDLINE" ]; then
+	# get UUID from fstab
+	uuid=$(awk '$2 == "/" { print $1 }' < /etc/fstab | sed 's/^UUID=//')
+	CMDLINE="root=UUID=$uuid quiet rw"
+fi
+
+mkdir -p "$entries"
+
+cat <<-EOF > "$entry"
+	title        Void Linux $ARCH $VERSION
+	sort-key     void
+	version      $VERSION
+	options      $CMDLINE
+	architecture $EFI_ARCH
+	linux        vmlinuz-$VERSION
+	initrd       initramfs-$VERSION.img
```

Comment:
It would be better if the hook confirmed that these files actually exist where expected, or else early abort. Better to have a bootloader entry for the last kernel persist than overwrite it with one that will not boot.

Reply:
add checks that these files actually exist, and will abort if they do not